six/modules/c++/six.sicd/tests/test_read_sicd_from_vts.cpp

```diff
@@ -0,0 +1,321 @@
+{
+public:
+    TestScript(const std::string& pathname) :
+        mPathname(pathname)
+    {
+        std::ifstream file(mPathname.c_str());
+        std::string line;
+        while (std::getline(file, line))
+        {
+            lines.push_back(line);
+        }
+        originalSource = str::split(lines[1], " ")[1];
+    }
+
+    ~TestScript()
+    {
+        setSource(originalSource);
+    }
+
+    void setSource(const std::string& sourcePathname)
+    {
+        std::vector<std::string> sourceLine = str::split(lines[1], " ");
+        sourceLine[1] = sourcePathname;
+        lines[1] = str::join(sourceLine, " ");
+        write();
+    }
+
+private:
+    std::vector<std::string> lines;
+    std::string originalSource;
+    const std::string mPathname;
+
+    void write()
+    {
+        std::ofstream file(mPathname.c_str());
+        for (size_t ii = 0; ii < lines.size(); ++ii)
+        {
+            file << lines[ii] << "\n";
+        }
+    }
+};
+
+class ResultFile
+{
+public:
+    ResultFile();
+    ~ResultFile();
+    std::string pathname() const;
+    bool operator==(const ResultFile& rhs) const;
+private:
+    std::string mName;
+};
+
+ResultFile::ResultFile() :
+    mName(std::tmpnam(NULL))
+{
+}
+
+ResultFile::~ResultFile()
+{
+    std::remove(mName.c_str());
+}
+
+std::string ResultFile::pathname() const
+{
+    return mName;
+}
+
+bool ResultFile::operator==(const ResultFile& rhs) const
+{
+    std::string leftLine;
+    std::string rightLine;
+    std::ifstream leftFile(pathname().c_str());
+    std::ifstream rightFile(rhs.pathname().c_str());
+
+    while (std::getline(leftFile, leftLine) &&
+        std::getline(rightFile, rightLine))
+    {
+        str::trim(leftLine);
+        str::trim(rightLine);
+        if (str::startsWith(leftLine, "Program End Time"))
+        {
+            break;
+        }
+
+        if (str::startsWith(leftLine, "Program Start Time") ||
+            str::endsWith(leftLine, ".nitf"))
+        {
+            continue;
+        }
+
+        if (leftLine != rightLine)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+}
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        validateArguments(argc, argv);
+        const std::string sicdPathname(argv[1]);
+        const std::string outputPathname(argv[2]);
+
+        attachDESs(sicdPathname, outputPathname);
+        const std::string delimiter(sys::Path::delimiter());
+        const std::string testFile = sys::Path::joinPaths("six",
+            sys::Path::joinPaths("projects",
+                sys::Path::joinPaths("csm",
+                    sys::Path::joinPaths("tests", "test_script_sicd"))));
+        TestScript testScript(testFile);
+        testScript.setSource(sicdPathname);
+        std::string vts = sys::Path::joinPaths("install",
+                sys::Path::joinPaths("bin", "vts"));
+        sys::OS os;
+        if (!os.isFile(vts))
+        {
+            vts = vts + ".exe";
+        }
+
```

Comment:
I was able to find SICD 0.4.1 and SIDD 1.0.0 NITFs in V:ComSAR\, but I’m having trouble finding anything for SICD 1.0. Do you know where I could find one?

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Friday, July 22, 2016 6:12 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Fix how CSM figures out an images type, to match how NITFReadControl now does it. (#90)


In six/modules/c++/six.sicd/tests/test_read_sicd_from_vts.cpp<https://github.com/ngageoint/six-library/pull/90#discussion_r71953436>:

> +        attachDESs(sicdPathname, outputPathname);

> +        const std::string delimiter(sys::Path::delimiter());

> +        const std::string testFile = sys::Path::joinPaths("six",

> +            sys::Path::joinPaths("projects",

> +                sys::Path::joinPaths("csm",

> +                    sys::Path::joinPaths("tests", "test_script_sicd"))));

> +        TestScript testScript(testFile);

> +        testScript.setSource(sicdPathname);

> +        std::string vts = sys::Path::joinPaths("install",

> +                sys::Path::joinPaths("bin", "vts"));

> +        sys::OS os;

> +        if (!os.isFile(vts))

> +        {

> +            vts = vts + ".exe";

> +        }

> +

Can you add tests for SICD 0.4 (that'll have SICD_XML DESID)?

Also a test for SIDD 1.0 (don't actually worry about SIDD_XML DESID... that was some really old version of SIDD that's not supported). We do want to make sure SIDD 1.0 works though.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ngageoint/six-library/pull/90/files/b5fa7dce12abfbd03f84ce03aee18d4669cf64de#r71953436>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYHUFHMsCh3Yr5JWfqU-Apfzj1LRDks5qYUASgaJpZM4JNnoK>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.